src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -207,8 +202,9 @@ public bool TargetFrameworkExists (string targetFramework)
 
 		public string Root {
```

Comment:
Can we remove this property? It doesn't appear to be used from anything.

Reply:
Xamarin.ProjectTools is not meant to be Xamarin.Android (even Xamarin.Android.Build.Tests) specific. Being unused from within that specific test assembly does not mean it is useless.

Reply:
I am completely happy to remove this project from Xamarin.Android and force you to submodule it if that doesn't match your expectation. It's almost MY project.

Reply:
@atsushieno: then we have a bit of a conundrum: we don't know what else is using `Xamarin.ProjectTools`; our knowledge is restricted to what we know about (xamarin-android, monodroid, their dependencies).

*This entire PR* will in all likelihood *break* whatever external project/repo you have which is using `src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools`, due to the new dependency for e.g. `bin/TestDebug/XABuildPaths.cs`. Without knowing what your specific project is, it's hard to be sure of anything.

> I am completely happy to remove this project from Xamarin.Android and force you to submodule it if that doesn't match your expectation

That may be for the best, to ensure that we don't break things.

Reply:
Addition of `XABuildPaths.cs` makes no difference in context of this project. The sources (e.g. https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs#L64-L114) make assumptions about locations that work *only* if the project is used from within XA tree and *only* if the test dll is found in a specific location. So the project is "non-portable" as-is. Addition of `XABuildPaths.cs` makes it better since the source then uses absolute path to the XA root and we get more portability. Whatever external project uses this assembly *must* be sourcing it from the XA tree already or else the Builder (and XamarinProject) would *not* work anyway. The only difference for any external project using it would be now to make sure XA tree is prepared (i.e. after `make prepare`) so that `XABuildPaths.cs` is generated. Putting `Xamarin.ProjectTools` in a submodule will break it *anyway* and similar measures to what this PR will be necessary to make it work (either that or parametrization of the classes so that the root path is passed to them). Either way, if you want to move this project to a submodule, please make it portable. This PR has nothing to do with it.

Reply:
We made a ton of changes to `Xamarin.ProjectTools` since it was first added. We added DotNetStandard support , solution support and a ton of other stuff. The Builder is already x-a specific since it looks for `xabuild.exe` and other x-a related stuff.

Reply:
That's true and they mostly don't break things too much. I can always replace it. Anyhow I'm not going to object the original changes unless RootPath gets removed, just to make sure.

Reply:
I left `Root` in because I haven't tested the changes in monodroid and I assumed that leaving it as-is will keep `monodroid` in a working state.